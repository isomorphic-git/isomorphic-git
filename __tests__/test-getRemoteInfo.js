/* eslint-env node, browser, jasmine */
// @ts-ignore
const snapshots = require('./__snapshots__/test-getRemoteInfo.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')
const { E, getRemoteInfo } = require('isomorphic-git')

describe('getRemoteInfo', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('getRemoteInfo', async () => {
    const info = await getRemoteInfo({
      url: 'http://localhost:8888/test-dumb-http-server.git'
    })
    expect(info).not.toBeNull()
    expect(info.capabilities).not.toBeNull()
    expect(info.refs).not.toBeNull()
    expect(info.refs).toMatchSnapshot()
  })
  ;(process.browser ? it : xit)(
    'detects "dumb" HTTP server responses',
    async () => {
      let error = null
      try {
        await getRemoteInfo({
          url:
            'http://localhost:9876/base/__tests__/__fixtures__/test-dumb-http-server.git'
        })
      } catch (err) {
        error = err
      }
      expect(error).not.toBeNull()
      expect(error.code).toBe(E.RemoteDoesNotSupportSmartHTTP)
    }
  )
})
