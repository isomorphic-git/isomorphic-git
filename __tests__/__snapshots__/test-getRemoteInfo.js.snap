// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRemoteInfo getRemoteInfo 1`] = `
Object {
  "heads": Object {
    "master": "97c024f73eaab2781bf3691597bc7c833cb0e22f",
    "test": "5a8905a02e181fe1821068b8c0f48cb6633d5b81",
  },
}
`;
